backends/platform/sdl/amigaos/amigaos-main.cpp

```diff
@@ -24,13 +24,44 @@
 
 #if defined(__amigaos4__)
 
+#include "backends/fs/amigaos4/amigaos4-fs.h"
 #include "backends/platform/sdl/amigaos/amigaos.h"
 #include "backends/plugins/sdl/sdl-provider.h"
 #include "base/main.h"
 
 int main(int argc, char *argv[]) {
 
-	// Set up a stack cookie to avoid crashes due to too few stack set by users 
+	// The following will gather the application name and add the install path
+	// to a variable in AmigaOS4's ENV(ARC) system. It will be placed in AppPaths
+	// so that ScummVM can become AmiUpdate aware
+	const char *const appname = "ScummVM";
+
+	BPTR lock;
+	APTR oldwin;
+
+		// Obtain the lock to the home directory
+		if(( lock = IDOS->GetProgramDir() )) {
```

Comment:
Space after if. Also I think that usually we don't use a space after the opening parenthesis and before the closing one.
And why use two sets of parentheses? One would be enough.